### ISSUE TITLE
Error in start if a gem is installed with an offline-pack

### ISSUE THREAD

**Issue by andsel** [elastic/logstash#11325]
When a gem in installed from an offline-pack LS log an error, else if the same gem is installed directly then works.

Log line in case of failure:
```
logstash_node1 | [ERROR] 2019-11-13 11:28:43.132 [main] Logstash - java.lang.IllegalStateException: Logstash stopped processing because of an error: (LoadError) Java plugin 'logstash-filter-mydata_document_enrichment' does not contain a single jar file with the plugin's name and version
mydata-elastic_logstash_1 exited with code 1
```

- Version: 7.4
- Operating System: x
- Config File (if you have sensitive info, please remove it): 
```
bin/logstash -e "input { stdin {} } output { stdout {codec => rubydebug} }"
```
- Steps to reproduce:

1. install a java gem (`bin/logstash-plugin install /gem_repo/logstash-filter-mygem-1.0.0.gem `)
2. start LogStash and verify logs and its functionality (`bin/logstash -e "input { stdin {} } output { stdout {codec => rubydebug} }"`)
3. generate an offlinepack (`bin/logstash-plugin prepare-offline-pack --output /tmp/logstash-filter-mygem.zip logstash-filter-mygem`)
4. uninstall the gem (`bin/logstash-plugin uninstall logstash-filter-mygem`)
5. install the offline pack (`bin/logstash-plugin install file:///tmp/logstash-filter-mygem.zip`)
6. start Logstash and verify log and functionality, that this time should fail (`bin/logstash -e "input { stdin {} } output { stdout {codec => rubydebug} }"`)

**Comment by andsel** [comment]
I've found that in the case of plain gem, the `bin/logstash-plugin install` at the and uses the Ruby's gem package to install it (`LogStash::PluginManager::Install.execute`, https://github.com/elastic/logstash/blob/7fb01b7e3be2c948e384858fbf2f8648ecd3f2e0/lib/pluginmanager/install.rb#L190-L205) because it doesn't recognize as `LogStash::PluginManager::PackFetchStrategy.Uri` strategy  nor `LogStash::PluginManager::PackFetchStrategy.Repository` becasue is not found on remote repo. Instead the zip file is decompressed locally, but on startup it tries to load the `spec.files` to check dependencies, but the list is empty

**Comment by andsel** [comment]
When a java plugin is installed with `gem` then during startup of https://github.com/elastic/logstash/blob/7fb01b7e3be2c948e384858fbf2f8648ecd3f2e0/logstash-core/lib/logstash/plugins/registry.rb#L26-L30 the `spec` is of `Gem::Specification` 
_but_ 
when the java plugin is installed with `zip` then during the startup the `spec` is `Bundler::StubSpecification`

The problem is that in the second case the field `spec.files` is empty

**Comment by andsel** [comment]
The problem is that during the installation of zip (executed by class `PackInstaller::Local`) it's invoked the method `LogStash::PluginManager::GemInstaller::install` which executes `write_specification`. This write_specification write a new `gemspec` file (in a folder like `vendor/bundle/jruby/2.5.0/specifications/logstash-filter-mydata_document_enrichment-1.0.0.gemspec`) where the file's content is:
`
# -*- encoding: utf-8 -*-
# stub: logstash-filter-mydata_document_enrichment 1.0.0 ruby libvendor/jar-dependencies

Gem::Specification.new do |s|
  s.name = "logstash-filter-mydata_document_enrichment".freeze
  s.version = "1.0.0"

  s.required_rubygems_version = Gem::Requirement.new(">= 0".freeze) if s.respond_to? :required_rubygems_version=
  s.metadata = { "java_plugin" => "true", "logstash_group" => "filter", "logstash_plugin" => "true" } if s.respond_to? :metadata=
  s.require_paths = ["lib".freeze, "vendor/jar-dependencies".freeze]
  s.authors = ["Elasticsearch".freeze]
  s.date = "2019-11-13"
  s.description = "This gem is a Logstash plugin required to be installed on top of the Logstash core pipeline using $LS_HOME/bin/logstash-plugin install gemname. This gem is not a stand-alone program".freeze
  s.email = ["info@elastic.co".freeze]
  s.homepage = "http://www.elastic.co/guide/en/logstash/current/index.html".freeze
  s.licenses = ["Apache-2.0".freeze]
  s.rubygems_version = "2.7.9".freeze
  s.summary = "MydataDocumentEnrichment".freeze

  s.installed_by_version = "2.7.9" if s.respond_to? :installed_by_version

  if s.respond_to? :specification_version then
    s.specification_version = 4

    if Gem::Version.new(Gem::VERSION) >= Gem::Version.new('1.2.0') then
      s.add_runtime_dependency(%q<logstash-core-plugin-api>.freeze, [">= 1.60", "<= 2.99"])
      s.add_runtime_dependency(%q<jar-dependencies>.freeze, [">= 0"])
      s.add_development_dependency(%q<logstash-devutils>.freeze, [">= 0"])
    else
      s.add_dependency(%q<logstash-core-plugin-api>.freeze, [">= 1.60", "<= 2.99"])
      s.add_dependency(%q<jar-dependencies>.freeze, [">= 0"])
      s.add_dependency(%q<logstash-devutils>.freeze, [">= 0"])
    end
  else
    s.add_dependency(%q<logstash-core-plugin-api>.freeze, [">= 1.60", "<= 2.99"])
    s.add_dependency(%q<jar-dependencies>.freeze, [">= 0"])
    s.add_dependency(%q<logstash-devutils>.freeze, [">= 0"])
  end
end
`

This specification is later used by the registry which in case of `"java_plugin" => true` metadata lists the `files` attribute to find the jar files to reference, and here born the problem. Indeed editing this generated spec file adding the `files` attribute with full path to  the jar file LS starts without any complain. Another way is to edit the `Gemfile` don't use the specification's gemspec but use the one from the gem `vendor/bundle/jruby/2.5.0/gems/logstash-filter-mydata_document_enrichment-1.0.0/logstash-filter-mydata_document_enrichment.gemspec`

**Comment by andsel** [comment]
Solved by #11340
